universal-application-tool-0.0.1/app/services/cloud/StorageClient.java

```diff
@@ -0,0 +1,26 @@
+package services.cloud;
+
+import java.net.URL;
+
+/** Interface for working with Cloud storage. We currently support AWS S3 and Azure Blob storage. */
+public interface StorageClient {
+
+  /**
+   * Returns a URL that gives users temporary access to file storage. This URL is used to upload
+   * users' files to cloud storage.
+   *
+   * @param fileName The file to be uploaded to cloud storage.
+   */
+  URL getPresignedUrl(String fileName);
+
+  /**
+   * Creates and returns a request to upload a file to cloud storage.
+   *
+   * @param fileName The file to upload to cloud storage
+   * @param successRedirectActionLink Where a user should be redirected upon successful file upload.
+   */
+  StorageUploadRequest getSignedUploadRequest(String fileName, String successRedirectActionLink);
```

Comment:
The comments state that both of these methods are for uploading, but isn't `getPresignedUrl` for downloading?